Tasks/Common/ServiceFabricHelpers/Connect-ServiceFabricClusterFromServiceEndpoint.ps1

```diff
@@ -210,4 +206,35 @@ function Connect-ServiceFabricClusterFromServiceEndpoint
     {
         Trace-VstsLeavingInvocation $MyInvocation
     }
+}
+
+function Connect-ServiceFabricClusterAction
+{
+    param(
+        [Hashtable]
+        $ClusterConnectionParameters
+    )
+
+    $connectResult = $null
+
+    try
+    {
+        # Call a trial Connect-ServiceFabricCluster first so that ServiceFabric PS module gets loaded. Retry only if this connect fails.
+        $connectResult = Connect-ServiceFabricCluster @clusterConnectionParameters
+    }
+    catch [System.Fabric.FabricTransientException], [System.TimeoutException]
+    {
+        $connectAction = { Connect-ServiceFabricCluster @clusterConnectionParameters }
+        $connectResult = Invoke-ActionWithRetries -Action $connectAction `
```

Comment:
Why are we not using Invoke-ActionWithDefaultRetries. This code is redundant. 
Why can't we move Invoke-ActionWithDefaultRetries code to Tasks/Common/ServiceFabricHelpers.

Reply:
This is just one usage in this module. When I add more, I will move Invoke-ActionWithDefaultRetries to common.

Reply:
I don't foresee more usage of Invoke-ActionWithRetries in ServiceFabricHelper.

Reply:
Last comment and I will drop this.  This is just a suggestion, consider it or discard it!

This is adding unnecessary redundancy due to which management issues will come up in future i.e. if we will decide to modify values of RetryIntervalInSeconds and MaxTries  then we have to modify it at 2 different places.